src/typed.jl

```diff
@@ -1,43 +1,56 @@
-abstract type AbstractRequestParams{RESPONSE_T} end
-abstract type AbstractNotificationParams end
+abstract type AbstractMessageType end
 
-function parse_response end
+struct NotificationType{TPARAM} <: AbstractMessageType
+    method::String
+    dict2param::Function
+end
+
+struct RequestType{TPARAM,TR} <: AbstractMessageType
+    method::String
+    dict2param::Function
```

Comment:
Does this mean the RequestType holds the methods for converting to and from the Dict that represents the JSON format? Would this then require a significant refactoring of the LSP (i.e. all the Dict->DataType converters built by `@dict_readable` share the name of DataType

Reply:
That was my initial idea, but I removed that. The current design is such that every `RequestType` just links the method name with a type for the parameters and a type for the return value.

For the conversion from `Dict` to type it just passes the dict to the constructor of the type, and for the conversion from type to JSON it just relies on whatever mechanism there is in JSON (I think). I think that means it should all work with the existing design for conversions, right? That was the goal, in any case ;)